# **Used Car Prices**

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
path = '/content/drive/MyDrive/train.csv'

In [ ]:
udin = pd.read_csv(path)

In [ ]:
udin

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type     Mileage  \
0                 41000    Diesel       Manual      First  19.67 kmpl   
1                 46000    Petrol       Manual      First    13 km/kg   
2                 87000    Diesel       Manual      First  20.77 kmpl   
3                 40670    Diesel    Automatic     Second   15.2 kmpl   
4                 86999    Diesel       Manual      First  23.08 kmpl   
...                 ...       ...          ...        ...         ...   
5842              27365    Diesel       Manual      First   28.4 kmpl   
5843             100000    Diesel       Manual      First   24.4 kmpl   
5844              55000    Diesel       Manual     Second   14.0 kmpl   
5845              46000    Petrol       Manual      First   18.9 kmpl   
5846              47000    Diesel       Manual      First  25.44 kmpl   

       Engine      Power  Seats  New_Price  Price  
0     1582 CC  126.2 bhp    5.0        NaN  12.50  
1     1199 CC   88.7 bhp    5.0  8.61 Lakh   4.50  
2     1248 CC  88.76 bhp    7.0        NaN   6.00  
3     1968 CC  140.8 bhp    5.0        NaN  17.74  
4     1461 CC   63.1 bhp    5.0        NaN   3.50  
...       ...        ...    ...        ...    ...  
5842  1248 CC     74 bhp    5.0  7.88 Lakh   4.75  
5843  1120 CC     71 bhp    5.0        NaN   4.00  
5844  2498 CC    112 bhp    8.0        NaN   2.90  
5845   998 CC   67.1 bhp    5.0        NaN   2.65  
5846   936 CC   57.6 bhp    5.0        NaN   2.50  

[5847 rows x 14 columns]

## **Business Understanding**

### **Business Objective**

Dataset yang digunakan merupakan data mobil bekas dari platform Cars24 dengan variabel utama Price (harga dalam satuan lakhs/Mata Uang India). Setiap observasi merepresentasikan satu unit kendaraan dengan berbagai atribut seperti merek dan model, tahun produksi, jarak tempuh, jenis bahan bakar, transmisi, kapasitas dan tenaga mesin, jumlah pemilik sebelumnya, serta harga mobil saat baru.

Permasalahan yang menjadi fokus dalam proyek ini adalah proses penentuan harga mobil bekas yang masih berpotensi kurang akurat karena sering bergantung pada perkiraan. Kondisi tersebut dapat menyebabkan harga tidak sesuai dengan nilai kendaraan di pasar.

Tujuan analisis bisnis adalah mengidentifikasi permasalahan dalam penetapan harga, merumuskan kebutuhan akan pendekatan berbasis data, serta mendukung proses pengambilan keputusan yang lebih objektif dan terukur. Melalui pemanfaatan data yang tersedia, analisis ini diharapkan dapat meningkatkan ketepatan penentuan harga serta mendorong proses penjualan yang lebih efisien.

### **Assess Sitution**

Dataset ini berbentuk data tabular (baris dan kolom) dan memiliki 5847 baris serta lebih dari lima kolom, sehingga memenuhi kriteria untuk dilakukan analisis. Dataset terdiri dari kombinasi variabel numerik dan kategorikal yang relevan untuk memprediksi harga mobil bekas.

Tetapi, terdapat beberapa masalah yang perlu diperhatikan. Beberapa atribut memiliki nilai yang hilang (missing values), seperti pada bagian kapasitas mesin dan tenaga mesin. Selain itu, terdapat kolom yang masih mengandung satuan seperti “CC” dan “bhp”, sehingga perlu dikonversi menjadi numerik murni agar dapat dianalisis karena bersifat string. Variasi harga yang cukup besar juga berpotensi menimbulkan outlier yang dapat memengaruhi hasil model.

Kondisi ini menunjukkan bahwa dataset memerlukan tahap pembersihan dan persiapan sebelum dilakukan pemodelan.

### **Analytic Goals Project Plan**

Tujuan analitik dari ini adalah untuk mengembangkan model prediksi berbasis regresi untuk memperkirakan harga mobil bekas berdasarkan atribut kendaraan yang tersedia. Selain menghasilkan estimasi harga, analisis ini juga bertujuan untuk mengidentifikasi variabel-variabel yang memiliki pengaruh signifikan terhadap nilai jual mobil.

Untuk mencapai tujuan tersebut, tahapan yang akan dilakukan dimulai dengan eksplorasi dan pemahaman struktur data, kemudian dilanjutkan dengan proses pembersihan serta persiapan data, termasuk penanganan missing values dan konversi tipe data. Setelah data siap, dilakukan analisis eksploratif guna melihat pola dan hubungan antar variabel. Selanjutnya dibangun model regresi untuk memprediksi harga mobil, dan diakhiri dengan evaluasi kinerja model serta penarikan kesimpulan berdasarkan hasil yang diperoleh.